param.go

```diff
@@ -48,8 +48,8 @@ type param interface {
 	// This MAY panic if the param does not produce a single value.
 	Build(containerStore) (reflect.Value, error)
 
-	// DotNodes returns a slice of param(s) represented in dot.Nodes for the DOT-format graph.
-	DotNodes() []*dot.Node
+	// DotNodes returns a slice of dot.Param(s).
+	DotNodes() []*dot.Param
```

Comment:
Now that these methods are different for parameters and results, let's consider renaming them to `DotParams` and `DotResults`.